test/e2e/autotls/config/dnssetup/main.go

```diff
@@ -77,14 +79,47 @@ func createDNSRecord(dnsRecord *config.DNSRecord) error {
 	return config.ChangeDNSRecord(addition, svc, env.CloudDNSProject, env.DNSZone)
 }
 
-func waitForDNSRecordVisibleLocally(record *config.DNSRecord) error {
+func waitForDNSRecordVisible(record *config.DNSRecord) error {
+	nameservers, err := net.LookupNS(env.DomainName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup NS records for domain: %v", err)
+	}
+
 	return wait.PollImmediate(10*time.Second, 300*time.Second, func() (bool, error) {
-		ips, _ := net.LookupHost(record.Domain)
-		for _, ip := range ips {
-			if ip == record.IP {
-				return true, nil
+		for _, ns := range nameservers {
+			nsIP, err := net.LookupHost(ns.Host)
+			if err != nil {
+				return true, err
+			}
+			r := &net.Resolver{
+				PreferGo: true,
```

Comment:
Why?

Reply:
From my testing when this is false it doesn't use the dialer I create.

Reply:
Please comment, since it's unclear why wouldn't we use system resolver.

Reply:
You made me second-guess myself :) I just checked again and it does work when it's false. I'm not sure what I ran into before but I'll remove this.

Reply:
I haven't looked at the source, but using system resolver probably would exclude your dialer, since it'll be a os.Exec call most likely?